torch/autograd/profiler.py

```diff
@@ -233,17 +231,18 @@ def __init__(
                 use_kineto
             ), "Device-only events supported only with Kineto (use_kineto=True)"
 
-        if self.use_device == "cuda":
-            self.use_device = None
-            self.use_cuda = True
-
-        if self.use_device and self.use_device != _get_privateuse1_backend_name():
+        if self.use_device and self.use_device not in [
+                "cuda", "xpu", _get_privateuse1_backend_name()]:
             warn(f"{self.use_device} doesn't support profile.")
             self.use_device = None
 
-        if self.use_cuda and not torch.cuda.is_available():
+        if self.use_device == "cuda" and not torch.cuda.is_available():
             warn("CUDA is not available, disabling CUDA profiling")
-            self.use_cuda = False
+            self.use_device = None
+
+        if self.use_device == "xpu" and not torch.xpu.is_available(): # type: ignore[attr-defined]
+            warn("XPU is not available, disabling XPU profiling")
+            self.use_device = None
```

Comment:
please add a check if use_device is not a valid option here. 
``` if self.use_device not in VALID_USE_DEViCE_OPTIONS:
 ```

Reply:
Sure. Thank you.